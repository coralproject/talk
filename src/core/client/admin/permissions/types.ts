import { GQLUSER_ROLE_RL } from "coral-framework/schema";

export type PermissionMap<AbilityType extends string, ContextType> = {
  [abilityType in AbilityType]: {
    [role in GQLUSER_ROLE_RL]?: (context?: ContextType) => boolean;
  };
};
