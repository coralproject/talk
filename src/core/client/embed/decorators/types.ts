import pym from "pym.js";

export type CleanupCallback = () => void;
export type Decorator = (pym: pym.Parent) => CleanupCallback | void;
