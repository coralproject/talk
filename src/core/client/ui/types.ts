import React from "react";

// TODO: Extract useful common types into its own package.

/**
 * Overwrite properties of `T`.
 *
 * E.g. Omit<{a: boolean, b: boolean}, "b"> = {a: boolean}
 */
export type Omit<U, K extends keyof U> = Pick<U, Exclude<keyof U, K>>;

/**
 * Overwrite properties of `T`.
 *
 * E.g. Overwrite<{a: boolean}, {a: string}> = {a: string}
 */
export type Overwrite<T, U> = Pick<T, Exclude<keyof T, keyof U>> & U;

/**
 * Returns the PropTypes from a React Component.
 *
 * E.g. type ButtonProps = ReturnPropTypes<Button>;
 */
export type PropTypesOf<T> = T extends React.ComponentType<infer R>
  ? R
  : T extends React.Component<infer S> ? S : {};
