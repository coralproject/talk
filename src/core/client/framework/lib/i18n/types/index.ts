export { default as FluentShortNumber } from "./FluentShortNumber";
