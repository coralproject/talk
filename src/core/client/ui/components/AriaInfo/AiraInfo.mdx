---
name: AriaInfo
menu: UI Kit
---

import { Playground, PropsTable } from 'docz'
import AriaInfo from './AriaInfo'

# AriaInfo

Renders Text that is only seen by screen readers.

## Basic usage
<Playground>
  <AriaInfo>This can only be seen by screen readers</AriaInfo>
  <div>Above this line is a hidden text that you can't see</div>
</Playground>
