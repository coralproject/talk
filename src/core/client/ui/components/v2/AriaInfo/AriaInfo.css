.root {
  position: absolute !important;
  width: 1px;
  height: 1px;
  padding: 0;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  white-space: nowrap;
  border: 0;
}
