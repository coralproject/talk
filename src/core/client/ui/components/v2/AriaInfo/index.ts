export { default, AriaInfoProps } from "./AriaInfo";
