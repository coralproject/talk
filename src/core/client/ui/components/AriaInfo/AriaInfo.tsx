import cn from "classnames";
import React, { HTMLAttributes, StatelessComponent } from "react";

import { withForwardRef, withStyles } from "talk-ui/hocs";
import { PropTypesOf } from "talk-ui/types";

import * as styles from "./AriaInfo.css";

interface InnerProps extends HTMLAttributes<HTMLElement> {
  /**
   * This prop can be used to add custom classnames.
   * It is handled by the `withStyles `HOC.
   */
  classes: typeof styles;
  component?: string;
  children: React.ReactNode;
}

const AriaInfo: StatelessComponent<InnerProps> = props => {
  const { component, className, classes, ...rest } = props;
  const Component = component!;
  const rootClassName = cn(classes.root, className);
  return <Component className={rootClassName} {...rest} />;
};

AriaInfo.defaultProps = {
  component: "div",
};

const enhanced = withForwardRef(withStyles(styles)(AriaInfo));
export type AriaInfoProps = PropTypesOf<typeof enhanced>;
export default enhanced;
