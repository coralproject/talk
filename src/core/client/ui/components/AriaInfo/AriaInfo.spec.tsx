import React from "react";
import TestRenderer from "react-test-renderer";

import { PropTypesOf } from "talk-framework/types";

import AriaInfo from "./AriaInfo";

it("renders correctly", () => {
  const props: PropTypesOf<typeof AriaInfo> = {
    children: "This text is only for screen readers",
  };
  const renderer = TestRenderer.create(<AriaInfo {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});
