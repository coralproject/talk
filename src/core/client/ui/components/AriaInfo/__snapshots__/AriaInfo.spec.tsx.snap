// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="AriaInfo-root"
>
  This text is only for screen readers
</div>
`;
